tests/sentry/workflow_engine/migration_helpers/test_migrate_alert_rule.py

```diff
@@ -526,3 +513,114 @@ def test_update_metric_alert_resolve_threshold(self):
         resolve_detector_trigger.refresh_from_db()
 
         assert resolve_detector_trigger.comparison == 10
+
+    def test_dual_delete_migrated_alert_rule_trigger(self):
+        migrate_alert_rule(self.metric_alert, self.rpc_user)
+        data_conditions = migrate_metric_data_conditions(self.alert_rule_trigger_critical)
+        assert data_conditions is not None
+        detector_trigger, action_filter = data_conditions
+
+        detector_trigger_id = detector_trigger.id
+        action_filter_id = action_filter.id
+        dual_delete_migrated_alert_rule_trigger(self.alert_rule_trigger_critical)
+        assert not DataCondition.objects.filter(id=detector_trigger_id).exists()
+        assert not DataCondition.objects.filter(id=action_filter_id).exists()
+
+    def test_dual_delete_migrated_alert_rule_trigger_action(self):
+        migrate_alert_rule(self.metric_alert, self.rpc_user)
+        migrate_metric_data_conditions(self.alert_rule_trigger_critical)
+        migrate_metric_action(self.alert_rule_trigger_action_integration)
+
+        aarta = ActionAlertRuleTriggerAction.objects.get(
+            alert_rule_trigger_action=self.alert_rule_trigger_action_integration
+        )
+        aarta_id = aarta.id
+        action_id = aarta.action.id
+        dual_delete_migrated_alert_rule_trigger_action(self.alert_rule_trigger_action_integration)
+        assert not Action.objects.filter(id=action_id).exists()
+        assert not ActionAlertRuleTriggerAction.objects.filter(id=aarta_id).exists()
+
+    def test_dual_delete_unmigrated_alert_rule_trigger(self):
+        """
+        Test that nothing weird happens if we try to dual delete a trigger whose alert rule was
+        never dual written.
+        """
+        # returns None quietly—nothing to assert, just make sure we don't throw an error
+        dual_delete_migrated_alert_rule_trigger(self.alert_rule_trigger_critical)
```

Comment:
should this assert that the result is `None` based on the comment? i think we should have some kind of assertion to validate it worked as expected vs not. (if i change the result of dual_delete_migrated_alert_rule_trigger this test would still pass, even though it's not returning none)